apps/lib/apps.c

```diff
@@ -1430,14 +1430,21 @@ BIGNUM *load_serial(const char *serialfile, int create, ASN1_INTEGER **retai)
 
     in = BIO_new_file(serialfile, "r");
     if (in == NULL) {
-        if (!create) {
+        if (create == NULL || !*create) {
             perror(serialfile);
             goto err;
         }
+        (*create)--;
```

Comment:
`*create = 0;`

Reply:
Ah, now I understand the dirty trick. I am not sure this is the right way to do it. I'd prefer keeping load_serial as is and just checking with BIO_new_file() whether the default file exists directly in x509_load_serial.

Reply:
Ok, I'll implement this in a more explicit way.
I do not like calling `BIO_new_file()` again because this is inefficient can requires restoring the error queue again.
Yet I can add an output parameter for passing down the file existence information.
On this occasion will rename `load_serial()` to `load_hex_number()` because in `apps.c` it is also used for CRL numbers.

Reply:
Done. Also renamed `save_serial()` to `save_hex()`.

Reply:
I am not sure the rename does not confuse things even more.